samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -732,89 +677,116 @@ static bool find_valid_bis_in_subgroup_bis(const struct bt_bap_base_subgroup *su
  *    configuration overwrites the subgroup values
  *
  * This function returns `true` if the subgroup does not support the channels in
- * CONFIG_TARGET_BROADCAST_CHANNEL which makes it iterate over the next subgroup, and returns
- * `false` if this subgroup satisfies our CONFIG_TARGET_BROADCAST_CHANNEL.
+ * CONFIG_TARGET_BROADCAST_CHANNEL which makes it iterate over the next subgroup, and
+ * returns `false` if this subgroup satisfies our CONFIG_TARGET_BROADCAST_CHANNEL.
  */
-static bool find_valid_bis_in_subgroup_cb(const struct bt_bap_base_subgroup *subgroup,
-					  void *user_data)
+static bool subgroup_get_valid_bis_indexes_cb(const struct bt_bap_base_subgroup *subgroup,
+					      void *user_data)
 {
+	enum bt_audio_location subgroup_chan_allocation = BT_AUDIO_LOCATION_MONO_AUDIO;
+	enum bt_audio_location bis_chan_allocation = BT_AUDIO_LOCATION_MONO_AUDIO;
```

Comment:
The concept of having a single `bis_chan_allocation` for the subgroup is kind of weird. 

For example later we have
```
chan_allocation = bis_chan_allocation;
if (chan_allocation == CONFIG_TARGET_BROADCAST_CHANNEL) {
```

But if `CONFIG_TARGET_BROADCAST_CHANNEL` == `LEFT | RIGHT` and `bis_chan_allocation == LEFT | RIGHT | FRONT_CENTER` when number of BIS are `3`, then the check fails even if the subgroup has a LEFT and RIGHT bis that we are looking for

Reply:
Right. I will change the code